File: 02/1.hs
main = do
    text <- readFile "input.txt"
    let input = map parseLine $ lines text
    let result = run input
    --putStrLn $ show $ take 5 input
    putStrLn $ show result

type Line = ((Int,Int),Char,String)

parseLine :: String -> Line
parseLine xs = ((read mi, read ma), ch, pw)
    where [limits, (ch:_), pw] = words xs
          (mi,(_:ma)) = span (/='-') limits

run :: [Line] -> Int
run = length . filter p
    where p ((mi,ma), ch, pw) =
            let s = length $ filter (==ch) pw in mi <= s && s <= ma

File: 02/1.js
const input = require("fs").readFileSync(process.argv[2] || "input.txt").toString().split("\n")
const lines = input.map(line => {
    let [limits, ch, password] = line.split(" ")
    limits = limits.split("-")
    ch = ch[0]
    const chSize = [...password].filter(c => c == ch).length
    const valid = chSize >= + limits[0] && chSize <= limits[1]
    return [valid, limits, ch, password]
})

console.log(lines.filter(l => l[0]).length)
